platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,255 @@
+# Merkle State API
+
+## Summary
+
+Refactor the Merkle tree so that it can be fully represented by the State API objects.
+Create a module that will provide a simple and easy-to-use API for the Merkle tree.
+
+## Purpose and Context
+
+### Why?
+
+The Merkle tree is a rather complex data structure, and to make interaction easier, we need to hide this complexity behind the API.
+Another reason to have this API is to provide additional flexibility in changing the implementation details of the Merkle tree.
+As long as they are hidden, we are free to change them. This API will be critically important for the Block Node to 
+interact with the state without knowing the details of the Merkle tree.
+
+Note that this API is already implemented as a part of the Service Modularization project, but these classes will be moved
```

Comment:
For me, it is important to see the dependencies of the API & Implementation. What additional modules/libs are needed?

Reply:
Sure. I specified all the dependencies